src/app.cpp

```diff
@@ -116,6 +116,37 @@ class CanFrame {
 
 };
 
+template<class ObjectHeaderGeneric>
+std::uint64_t calculate_ts_res(ObjectHeaderGeneric *oh)
```

Comment:
Common logic with write_packet, duplicate block?

Reply:
Yes. I have copy-pasted the functionality for LIN and have now used this function also in write_packet. It will be reflected in the next commit.